projection/raw_transaction/raw_transaction.go

```diff
@@ -0,0 +1,208 @@
+
+func (_ *RawTransaction) GetEventsToListen() []string {
+	return []string{
+		event_usecase.BLOCK_CREATED,
+		event_usecase.RAW_TRANSACTION_CREATED,
+		event_usecase.RAW_TRANSACTION_FAILED,
+	}
+}
+
+func (projection *RawTransaction) OnInit() error {
+	if projection.migrationHelper != nil {
+		projection.migrationHelper.Migrate()
+	}
+
+	return nil
+}
+
+func (projection *RawTransaction) HandleEvents(height int64, events []event_entity.Event) error {
+	rdbTx, err := projection.rdbConn.Begin()
+	if err != nil {
+		return fmt.Errorf("error beginning transaction: %v", err)
+	}
+
+	committed := false
+	defer func() {
+		if !committed {
+			_ = rdbTx.Rollback()
+		}
+	}()
+
+	rdbTxHandle := rdbTx.ToHandle()
+	rawTransactionsView := NewRawTransactions(rdbTxHandle)
+	rawTransactionsTotalView := NewRawTransactionsTotal(rdbTxHandle)
+
+	var blockTime utctime.UTCTime
+	var blockHash string
+	txs := make([]raw_transaction_view.RawTransactionRow, 0)
+	for _, event := range events {
+		if blockCreatedEvent, ok := event.(*event_usecase.BlockCreated); ok {
+			blockTime = blockCreatedEvent.Block.Time
+			blockHash = blockCreatedEvent.Block.Hash
+		} else if transactionCreatedEvent, ok := event.(*event_usecase.RawTransactionCreated); ok {
+			tx := raw_transaction_view.RawTransactionRow{
+				BlockHeight:   height,
+				BlockTime:     utctime.UTCTime{}, // placeholder
+				Hash:          transactionCreatedEvent.TxHash,
+				Index:         transactionCreatedEvent.Index,
+				Success:       true,
+				Code:          transactionCreatedEvent.Code,
+				Log:           transactionCreatedEvent.Log,
+				Fee:           transactionCreatedEvent.Fee,
+				FeePayer:      transactionCreatedEvent.FeePayer,
+				FeeGranter:    transactionCreatedEvent.FeeGranter,
+				GasWanted:     transactionCreatedEvent.GasWanted,
+				GasUsed:       transactionCreatedEvent.GasUsed,
+				Memo:          transactionCreatedEvent.Memo,
+				TimeoutHeight: transactionCreatedEvent.TimeoutHeight,
+				Messages:      transactionCreatedEvent.Messages,
+			}
+
+			signers := make([]raw_transaction_view.RawTransactionRowSigner, 0)
+			for _, transactionCreatedEventSigner := range transactionCreatedEvent.Signers {
+				transactionRowSigner := raw_transaction_view.RawTransactionRowSigner{
+					AccountSequence: transactionCreatedEventSigner.AccountSequence,
+					Address:         transactionCreatedEventSigner.Address,
+				}
+				if transactionCreatedEventSigner.MaybeKeyInfo != nil {
+					transactionRowSigner.MaybeKeyInfo = &raw_transaction_view.RawTransactionRowSignerKeyInfo{
+						Type:           transactionCreatedEventSigner.MaybeKeyInfo.Type,
+						IsMultiSig:     transactionCreatedEventSigner.MaybeKeyInfo.IsMultiSig,
+						Pubkeys:        transactionCreatedEventSigner.MaybeKeyInfo.Pubkeys,
+						MaybeThreshold: transactionCreatedEventSigner.MaybeKeyInfo.MaybeThreshold,
+					}
+				}
+				signers = append(signers, transactionRowSigner)
+			}
+			tx.Signers = signers
+			txs = append(txs, tx)
```

Comment:
I remember we have a conversation about signer info maybe missing in some cases. Can you shared again the conclusion? Is it that the signer always present?

Reply:
empty signer would be exist here since we do not process any data inside the messages